parlai/agents/transformer/modules/encoder.py

```diff
@@ -141,7 +218,7 @@ def _default(val, default):
         self.layers = nn.ModuleList()
         for _ in range(self.n_layers):
             self.layers.append(
-                TransformerEncoderLayer(
+                template.layer.build(
```

Comment:
this is really hard to read.. can't it be called encoderlayer_class or something (assuming that's what it is)?  template is so generic...

Reply:
That's pretty much what it was called in the previous commit lol. The more I stare at and play with the code the less sure I am about what's intuitive and what's confusing.